superset/utils/dashboard_filter_scopes_converter.py

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+import logging
+from collections import defaultdict
+from typing import Dict, List
+
+from superset.models.slice import Slice
+
+logger = logging.getLogger(__name__)
+
+
+def convert_filter_scopes(json_metadata: Dict, filters: List[Slice]):
+    filter_scopes = {}
+    immuned_by_id: List[int] = json_metadata.get("filter_immune_slices") or []
+    immuned_by_column: Dict = defaultdict(list)
+    for slice_id, columns in json_metadata.get(
+        "filter_immune_slice_fields", {}
+    ).items():
+        for column in columns:
+            immuned_by_column[column].append(int(slice_id))
+
+    def add_filter_scope(filter_field, filter_id):
+        # in case filter field is invalid
+        if isinstance(filter_field, str):
+            current_filter_immune = list(
+                set(immuned_by_id + immuned_by_column.get(filter_field, []))
+            )
+            filter_fields[filter_field] = {
+                "scope": ["ROOT_ID"],
+                "immune": current_filter_immune,
+            }
+        else:
+            logging.info(f"slice [{filter_id}] has invalid field: {filter_field}")
+
+    for filter_slice in filters:
+        filter_fields: Dict = {}
+        filter_id = filter_slice.id
+        slice_params = json.loads(filter_slice.params or "{}")
+        configs = slice_params.get("filter_configs") or []
+
+        if slice_params.get("date_filter"):
```

Comment:
How do we know this is the exhaustive list of slice parameters?

Reply:
![Screen_Shot_2020-02-14_at_10_29_54_AM](https://user-images.githubusercontent.com/27990562/74557777-c10d4000-4f15-11ea-9aa3-136db8667efc.jpg)


I was thinking about to add an iteration for all the possible date filter related keys.  But think this function will be used by:
- db migration, only run once,
- dashboard import, to convert `filter_immune` metadata in old dashboard to use new `filter_scopes`
Given the usage is kind of backward compatible not future compatible, I feel enumeration probably is good enough?